File: app/src/main/java/com/byoutline/kickmaterial/managers/CategoriesListViewModel.java
package com.byoutline.kickmaterial.managers;

import android.databinding.ObservableArrayList;
import android.support.annotation.Nullable;
import com.byoutline.kickmaterial.BR;
import com.byoutline.kickmaterial.R;
import com.byoutline.kickmaterial.dagger.GlobalScope;
import com.byoutline.kickmaterial.model.Category;
import me.tatarka.bindingcollectionadapter.ItemView;

import javax.inject.Inject;

@GlobalScope
public class CategoriesListViewModel {
    public final ObservableArrayList<Category> items = DataManager.getCategoriesList();
    public final ItemView itemView = ItemView.of(BR.categoryItem, R.layout.category_list_item);

    @Inject
    public CategoriesListViewModel() {
    }

    public void setAllCategoriesBgColor(@Nullable Integer color) {
        items.get(0).setBgColor(color);
    }
}

File: app/src/main/java/com/byoutline/kickmaterial/managers/CategoriesListViewModel.kt
package com.byoutline.kickmaterial.managers

import android.databinding.ObservableArrayList
import com.byoutline.kickmaterial.BR
import com.byoutline.kickmaterial.R
import com.byoutline.kickmaterial.dagger.GlobalScope
import com.byoutline.kickmaterial.model.Category
import me.tatarka.bindingcollectionadapter2.ItemBinding
import javax.inject.Inject

@GlobalScope
class CategoriesListViewModel @Inject
constructor() {
    val items: ObservableArrayList<Category> = DataManager.categoriesList
    val itemBinding = ItemBinding.of<Category>(BR.categoryItem, R.layout.category_list_item)

    fun setAllCategoriesBgColor(color: Int?) {
        items[0].setBgColor(color)
    }
}